mindocr/data/transforms/det_transforms.py

```diff
@@ -615,17 +603,15 @@ def __call__(self, data: dict):
                     poly = poly.intersection(border)
                     if poly.area < self._min_area:
                         ignore = True
-                    poly = poly.exterior
+                    poly = poly.exterior                    # fixme: poly.intersection could be a MultiPolygon
                     poly = poly.coords[::-1] if poly.is_ccw else poly.coords    # sort in clockwise order
                     new_polys.append(np.array(poly[:-1]))
-
                 else:                                       # the polygon is fully outside the image
                     continue
             new_tags.append(ignore)
             new_texts.append(text)
 
-        data['polys'] = new_polys
+        data['polys'] = np.array(new_polys)  # need to use numpy.ndarray
```

Comment:
ok

Reply:
Added, but ci failed with this 😥
<img width="1038" alt="截屏2023-06-02 10 43 23" src="https://github.com/mindspore-lab/mindocr/assets/80800595/6df5241b-7667-4da9-9ef9-2b882acd213d">

Reply:
This is because clipping polygons may result in different number of vertices. The new pipeline for `ValidatePolygons` is in PR #292. You can revert your changes in `ValidatePolygons` to avoid conflict with #292.

Reply:
Good suggestion, reverted.